ResearchKit/Common/ORKIconButton.m

```diff
@@ -0,0 +1,138 @@
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "ORKIconButton.h"
+#import "ORKHelpers_Internal.h"
+#import "ORKLabel.h"
+
+@implementation ORKIconButton {
+    UIView *_customView;
+    UIImageView *_imageView;
+    ORKLabel *_textLabel;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self init_ORKIconButton];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self init_ORKIconButton];
+    }
+    return self;
+}
+
+- (instancetype)initWithButtonText:(NSString *)buttonText buttonIcon:(UIImage *)buttonIcon {
+    self = [super init];
+    if (self) {
+        self.buttonText = buttonText;
+        self.buttonIcon = buttonIcon;
+        [self init_ORKIconButton];
+    }
+    return self;
+}
+
+- (void)init_ORKIconButton {
+    self.layer.cornerRadius = 10.0;
+    self.clipsToBounds = YES;
+    
+    [self setupCustomView];
+    [self setupTextLabel];
+    [self setupImageView];
+    [self setupConstraints];
+}
+
+- (void)setupCustomView {
+    _customView = [UIView new];
+    _customView.translatesAutoresizingMaskIntoConstraints = NO;
+    [_customView setBackgroundColor:[UIColor lightGrayColor]];
+    [_customView setUserInteractionEnabled:NO];
+    
+    [self addSubview:_customView];
+}
+
+- (void)setupTextLabel {
+    _textLabel = [ORKLabel new];
+    _textLabel.translatesAutoresizingMaskIntoConstraints = NO;
+    _textLabel.textAlignment = NSTextAlignmentLeft;
+    _textLabel.lineBreakMode = NSLineBreakByWordWrapping;
+    _textLabel.numberOfLines = 0;
+    _textLabel.text = !_buttonText ? @"" : _buttonText;
+    [_textLabel setTextColor:[UIColor systemBlueColor]];
+    UIFontDescriptor *descriptor = [UIFontDescriptor preferredFontDescriptorWithTextStyle:UIFontTextStyleFootnote];
+    UIFontDescriptor *fontDescriptor = [descriptor fontDescriptorWithSymbolicTraits:(UIFontDescriptorTraitBold)];
+    [_textLabel setFont:[UIFont fontWithDescriptor:fontDescriptor size:[[descriptor objectForKey: UIFontDescriptorSizeAttribute] doubleValue]]];
+    
+    [_customView addSubview:_textLabel];
+}
+
+- (void)setupImageView {
+    _imageView = [UIImageView new];
+    _imageView.image = _buttonIcon;
+    _imageView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    [_customView addSubview:_imageView];
+}
+
+- (void)setupConstraints {
+    [[_textLabel.centerYAnchor constraintEqualToAnchor:_customView.centerYAnchor] setActive:YES];
+    [[_textLabel.leadingAnchor constraintEqualToAnchor:_customView.leadingAnchor constant:15.0] setActive:YES];
```

Comment:
@srinathtm-apple forgot to add constants for the values here. Is that worth changing/adding right now?